discovery/kubernetes/client_metrics.go

```diff
@@ -136,6 +136,22 @@ var (
 		},
 		[]string{"queue_name"},
 	)
+	clientGoWorkqueueUnfinishedWorkSecondsMetricVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: workqueueMetricsNamespace,
+			Name:      "unfinished_work_seconds",
+			Help:      "How long an item has remained unfinished in the work queue.",
+		},
+		[]string{"queue_name"},
+	)
+	clientGoWorkqueueLongestRunningProcessorMetricVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: workqueueMetricsNamespace,
+			Name:      "longest_running_processor_micro_seconds",
```

Comment:
The metric in Prometheus should be in seconds. See how it is done with `clientGoWorkqueueLatencyMetricVec` for instance.

Reply:
@simonpasquier I am still researching on how to do this for the Gauge vector. ( i think I need to pass in a GaugeFunc somehow). The other places which do the conversion are for Summary metric.

Reply:
@simonpasquier any tips on this? I am still looking for a way to do this conversion before setting it to Gauge

Reply:
I haven't looked into the details. I'll do soon.

Reply:
You could create a type that would satisfy the `workqueue.SettableGaugeMetric` interface and do the conversion from microseconds to seconds like this:

```diff
diff --git a/discovery/kubernetes/client_metrics.go b/discovery/kubernetes/client_metrics.go
index 19dc7023..6d76a74d 100644
--- a/discovery/kubernetes/client_metrics.go
+++ b/discovery/kubernetes/client_metrics.go
@@ -258,8 +258,17 @@ func (f *clientGoWorkqueueMetricsProvider) NewWorkDurationMetric(name string) wo
                metric.Observe(v / 1e6)
        })
 }
+
+type settableGaugeFunc func(float64)
+
+func (f settableGaugeFunc) Set(value float64) {
+       f(value)
+}
 func (f *clientGoWorkqueueMetricsProvider) NewUnfinishedWorkSecondsMetric(name string) workqueue.SettableGaugeMetric {
-       return clientGoWorkqueueUnfinishedWorkSecondsMetricVec.WithLabelValues(name)
+       m := clientGoWorkqueueUnfinishedWorkSecondsMetricVec.WithLabelValues(name)
+       return settableGaugeFunc(func(v float64) {
+               m.Set(v / 1e6)
+       })
 }
 func (f *clientGoWorkqueueMetricsProvider) NewLongestRunningProcessorMicrosecondsMetric(name string) workqueue.SettableGaugeMetric {
        return clientGoWorkqueueLongestRunningProcessorMetricVec.WithLabelValues(name)
```

Reply:
Yes. That was what had come to mind. I was wondering if that was okay. Is there a particular location I should declare this type at (similar types seemed to be declared in the client_golang repo)? Or is it to declare the type in `kubernetes\client_metrics.go` ?

Reply:
Having it in `client_metrics.go` is ok.

Reply:
Thanks:)